core/local/atom_watcher.js

```diff
@@ -0,0 +1,86 @@
+/* @flow */
+
+const fse = require('fs-extra')
+const Promise = require('bluebird')
+
+const checksumer = require('./checksumer')
+const logger = require('../logger')
+const LinuxSource = require('./layers/linux')
+const Identifier = require('./layers/identifier')
+const ChecksumLayer = require('./layers/checksum')
+const Dispatcher = require('./layers/dispatcher')
+
+/*::
+import type Pouch from '../pouch'
+import type Prep from '../prep'
+import type EventEmitter from 'events'
+import type { Checksumer } from './checksumer'
+import type { ChokidarEvent } from './chokidar_event'
+*/
+
+const log = logger({
+  component: 'AtomWatcher'
+})
+
+module.exports = class AtomWatcher {
+  /*::
+  syncPath: string
+  prep: Prep
+  pouch: Pouch
+  events: EventEmitter
+  checksumer: Checksumer
+  ensureDirInterval: *
+  running: Promise<void>
+  _runningResolve: ?Function
+  _runningReject: ?Function
+  source: LinuxSource
+  start: () => Promise<*>
+  stop: (force: ?bool) => Promise<*>
+  ensureDirSync: () => void
+  */
+
+  constructor (syncPath /*: string */, prep /*: Prep */, pouch /*: Pouch */, events /*: EventEmitter */) {
+    this.syncPath = syncPath
+    this.prep = prep
+    this.pouch = pouch
+    this.events = events
+    this.checksumer = checksumer.init()
```

Comment:
`this.prep`, `this.pouch` and `this.checksumer` don't seem to be used outside of the constructor.

Reply:
`this.checksumer` is used from the outside: https://github.com/cozy-labs/cozy-desktop/blob/master/core/local/index.js#L236

I've removed the two others